homework/index.js

```diff
@@ -29,23 +31,154 @@
+      class: 'contributor-img',
+    });
+    createAndAppend('a', li, {
+      text: contributor.login,
+      href: contributor.html_url,
+      target: '_blank',
+    });
+    createAndAppend('div', li, {
+      text: contributor.contributions,
+      class: 'contributions-div',
+    });
+  }
+
+  function renderContribution(contributorsUrl, section) {
+    section.innerHTML = '';
+    const ul = createAndAppend('ul', section);
+    createAndAppend('p', ul, { text: 'Contributions' });
+
+    const contributorsPromise = fetchJSON(contributorsUrl);
+    contributorsPromise
+      .then(data => {
+        data.forEach(contributor => createListItem(ul, contributor));
+      })
+      .catch(err => {
+        createErrorMessage(err);
+      });
+  }
+
+  function selectRepository(select, mainDiv, input, initialSelectValue = '') {
+    return function() {
+      mainDiv.innerHTML = '';
+      const basicInfoSec = createAndAppend('section', mainDiv);
+      const contributionsSec = createAndAppend('section', mainDiv, {
+        class: 'contributions-section',
+      });
+      const selectedRepoIndex =
+        select.options[select.options.selectedIndex].value;
+
+      if (initialSelectValue === 0) {
+        renderRepoDetails(input, basicInfoSec);
+        renderContribution(input.contributors_url, contributionsSec);
+      } else {
+        renderRepoDetails(input[selectedRepoIndex], basicInfoSec);
+        renderContribution(
+          input[selectedRepoIndex].contributors_url,
+          contributionsSec,
+        );
       }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+    };
+  }
+
+  function renderRepoSelect(repo, optionValue, select, mainDiv) {
+    const repoItem = createAndAppend('option', select, {
+      text: repo.name,
+      value: optionValue,
+    });
+    if (optionValue === 0) {
+      repoItem.selected = true;
+      const selectFirstOption = selectRepository(
+        select,
+        mainDiv,
+        repo,
+        optionValue,
+      );
+      selectFirstOption();
+    }
+  }
+
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      class: 'main-header',
+    });
+    createAndAppend('div', header, {
+      class: 'header-text',
+      text: 'HYF-Repositories',
     });
+    const select = createAndAppend('select', header, {
+      class: 'repo-select',
+    });
+    const mainSection = createAndAppend('main', root, { id: 'main' });
+
+    const promise = fetchJSON(url);
+    promise
+      .then(repos => {
+        repos
+          .sort((firstRepo, secondRepo) => {
+            return firstRepo.name.localeCompare(secondRepo.name);
+          })
+          .forEach((repo, index) =>
+            renderRepoSelect(repo, index, select, mainSection),
+          );
+        select.addEventListener(
+          'change',
+          selectRepository(select, mainSection, repos),
+        );
```

Comment:
Yes, that's what I was looking for. How to call selectRepository at start-up since there is no change event to fire. Very nice feedback I really appreciate it. Thank you Jim.